internal/stage5.go

```diff
@@ -57,6 +58,15 @@ func testAPIVersion(stageHarness *test_case_harness.TestCaseHarness) error {
 	}
 	stageLogger.Debugf("Hexdump of received \"ApiVersions\" response: \n%v\n", GetFormattedHexdump(response.RawBytes))
 
+	messageSizeField := int(binary.BigEndian.Uint32(response.RawBytes[:4]))
+	receivedPayloadLength := len(response.Payload)
+
+	if messageSizeField != receivedPayloadLength {
```

Comment:
@andy1li this isn't unique to the ApiVersions test, it can apply to any response we read from the broker. Can we try to place it somewhere where it applies to all stages?